ygot/render.go

```diff
@@ -283,6 +455,40 @@ func leavesToNotifications(leaves map[*path]interface{}, ts int64, prefix []inte
 	return []*gnmipb.Notification{n}, nil
 }
 
+// stripPrefix removes the prefix pfx from the supplied path, and returns the more
+// specific path elements of the path. If the prefix is invalid for the path an
+// error is return.
+func stripPrefix(path *gnmiPath, pfx *gnmiPath) (*gnmiPath, error) {
+	if path.isStringSlicePath() != pfx.isStringSlicePath() {
+		return nil, fmt.Errorf("mismatched path formats in prefix and path, isElementPath: %v != %v", path.isStringSlicePath(), pfx.isStringSlicePath())
+	}
+
+	if !path.isValid() || !pfx.isValid() {
+		return nil, fmt.Errorf("invalid paths supplied for stripPrefix: %v, %v", path, pfx)
+	}
+
+	if pfx.isStringSlicePath() {
```

Comment:
This could be rewritten generically in terms of At() method e.g. 
for i := 0; i < pfx.Len(); i++ {
  if pfx.At(i) != path.At(i) { // no match }
}
return path.SubPath(i, path.Len())